src/log-store/src/raft_engine/log_store.rs

```diff
@@ -580,4 +575,36 @@ mod tests {
         vec.sort_by(|a, b| a.id.partial_cmp(&b.id).unwrap());
         assert_eq!(101, vec.first().unwrap().id);
     }
+
+    #[allow(clippy::mutable_key_type)]
+    #[tokio::test]
+    async fn test_append_batch() {
+        common_telemetry::init_default_ut_logging();
+        let dir = create_temp_dir("logstore-append-batch-test");
+
+        let config = LogConfig {
+            log_file_dir: dir.path().to_str().unwrap().to_string(),
+            file_size: ReadableSize::mb(2).0,
+            purge_threshold: ReadableSize::mb(4).0,
+            purge_interval: Duration::from_secs(5),
+            ..Default::default()
+        };
+
+        let logstore = RaftEngineLogStore::try_new(config).await.unwrap();
+
+        let entries = (0..8)
+            .flat_map(|ns_id| {
```

Comment:
Do we need to add another test case for interleaving different regions' entries?

e.g.
```
[
(ns1, data1-1),
(ns2, data2-1),
(ns1, data1-2),
(ns2, data2-2),
]
```